internal/db/schema/statement.go

```diff
@@ -0,0 +1,59 @@
+package schema
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// statementProvider provides the migration statements in order.
+// Next should be called prior to calling Version() or ReadUp() or sentinel
+// values (-1 and nil) will be returned.
+type statementProvider struct {
+	pos      int
+	versions []int
+	up, down map[int][]byte
+}
+
+func newStatementProvider(dialect string, curVer int) (*statementProvider, error) {
+	op := errors.Op("schema.newStatementProvider")
+	qp := statementProvider{pos: -1}
+	qp.up, qp.down = migrationStates[dialect].upMigrations, migrationStates[dialect].downMigrations
```

Comment:
should there be a function to get these from the package var?  Sort of like DevMigration(...) or BinarySchemaVersion(...)

Reply:
Added functions for both up and down.